packages/jsii/lib/assembler.ts

```diff
@@ -311,13 +327,105 @@ export class Assembler implements Emitter {
     }
   }
 
+  private _registerNamespaces(symbol: ts.Symbol): void {
+    const declaration = symbol.valueDeclaration ?? symbol.declarations[0];
+    if (declaration == null || !ts.isNamespaceExport(declaration)) {
+      // Nothing to do here...
+      return;
+    }
+    const moduleSpecifier = declaration.parent.moduleSpecifier;
+    if (moduleSpecifier == null || !ts.isStringLiteral(moduleSpecifier)) {
+      // There is a grammar error here, so we'll let tsc report this for us.
+      return;
+    }
+    const resolution = ts.resolveModuleName(
+      moduleSpecifier.text,
+      declaration.getSourceFile().fileName,
+      this.program.getCompilerOptions(),
+      ts.sys
+    );
+    if (resolution.resolvedModule == null) {
+      // Unresolvable module... We'll let tsc report this for us.
+      return;
+    }
+    if (resolution.resolvedModule.isExternalLibraryImport) {
+      // External re-exports are "pure-javascript" sugar; they need not be
+      // represented in the jsii Assembly since the types in there will be
+      // resolved through dependencies.
+      return;
```

Comment:
Is this something we want to report/warn?

Reply:
I don't think so. I feel it's a legitimate use-case, and the fact we need not represent those in *jsii*-land is actually just an implementation detail... Of course this is subject to change if we discover this is surprising in some way.